specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json

```diff
@@ -12605,17 +12719,88 @@
+        },
+        "licenseType": {
+          "type": "string",
+          "title": "If the the image or disk that is being used was licensed on-premises.",
+          "description": "This only applies to images that contain the Windows operating system. Values are:\n\n Windows_Server - The on-premises license is for Windows Server.\n Windows_Client - The on-premises license is for Windows Client."
+        },
+        "containerConfiguration": {
+          "$ref": "#/definitions/ContainerConfiguration",
+          "title": "The container configuration for the pool.",
+          "description": "If this is specified, then all regular tasks and job manager tasks run on this pool must specify the containerSettings property."
         }
       },
-      "required": [ "nodeAgentSKUId" ],
+      "required": [ "nodeAgentSKUId", "imageReference" ],
       "title": "The configuration for compute nodes in a pool based on the Azure Virtual Machines infrastructure."
     },
+    "ContainerRegistry": {
+      "properties": {
+        "registryServer": {
+          "type": "string",
+          "title": "The registry server url.",
+          "description": "If omitted, the default is: \"docker.io\"."
+        },
+        "username": {
+          "type": "string",
+          "x-ms-client-name": "userName",
+          "title": "The user name to login to the registry server."
+        },
+        "password": {
+          "type": "string",
+          "title": "The password to login to the registry server."
+        }
+      },
+      "required": [ "username", "password" ],
+      "title": "Private container registry details."
+    },
+    "ContainerConfiguration": {
+      "properties": {
+        "type": {
+          "type": "string",
+          "title": "Which container technology will be used.",
+          "description": "Values are:\n\n docker - The container will use Docker.",
+          "enum": [
+            "docker"
+          ],
+          "x-ms-enum": {
+            "name": "ContainerType",
+            "modelAsString": false
+          }
+        },
+        "containerImageNames": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "title": "The collection of container image names.",
+          "description": "This is the full image reference, as would be specified to \"docker pull\". An image will be sourced from the default Docker registry unless the image is fully qualified with an alternative registry."
+        },
+        "containerRegistries": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ContainerRegistry"
+          },
+          "title": "Additional private registries which containers can be pulled from."
+        }
+      },
+      "required": [ "type" ],
+      "title": "The configuration for container-enabled compute nodes."
```

Comment:
Compute nodes or pools?  Also is 'enabled' right because I thought this meant containerisation was _enforced_ not just an option.

Reply:
Yeah, pools makes more sense here.